spring-web/src/main/java/org/springframework/http/server/reactive/JettyCoreServerHttpResponse.java

```diff
@@ -0,0 +1,315 @@
+		}
+		catch (Throwable th) {
+			callback.failed(th);
+		}
+		return doCommit(() -> mono);
+	}
+
+	private Mono<Void> sendDataBuffer(DataBuffer dataBuffer) {
+		@SuppressWarnings("resource")
+		DataBuffer.ByteBufferIterator byteBufferIterator = dataBuffer.readableByteBuffers();
+		Callback.Completable callback = new Callback.Completable();
+		new IteratingCallback() {
+			@Override
+			protected Action process() {
+				if (!byteBufferIterator.hasNext()) {
+					return Action.SUCCEEDED;
+				}
+				response.write(false, byteBufferIterator.next(), this);
+				return Action.SCHEDULED;
+			}
+
+			@Override
+			protected void onCompleteSuccess() {
+				byteBufferIterator.close();
+				DataBufferUtils.release(dataBuffer);
+				callback.complete(null);
+			}
+
+			@Override
+			protected void onCompleteFailure(Throwable cause) {
+				byteBufferIterator.close();
+				DataBufferUtils.release(dataBuffer);
+				callback.failed(cause);
+			}
+		}.iterate();
+
+		return doCommit(() -> Mono.fromFuture(callback));
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public <T> T getNativeResponse() {
+		return (T) this.response;
+	}
+
+	private static class ResponseHttpCookie implements org.eclipse.jetty.http.HttpCookie {
+		private final ResponseCookie responseCookie;
+
+		public ResponseHttpCookie(ResponseCookie responseCookie) {
+			this.responseCookie = responseCookie;
+		}
+
+		public ResponseCookie getResponseCookie() {
+			return this.responseCookie;
+		}
+
+		@Override
+		public String getName() {
+			return this.responseCookie.getName();
+		}
+
+		@Override
+		public String getValue() {
+			return this.responseCookie.getValue();
+		}
+
+		@Override
+		public int getVersion() {
+			return 0;
+		}
+
+		@Override
+		public long getMaxAge() {
+			return this.responseCookie.getMaxAge().toSeconds();
+		}
+
+		@Override
+		@Nullable
+		public String getComment() {
+			return null;
+		}
+
+		@Override
+		@Nullable
+		public String getDomain() {
+			return this.responseCookie.getDomain();
+		}
+
+		@Override
+		@Nullable
+		public String getPath() {
+			return this.responseCookie.getPath();
+		}
+
+		@Override
+		public boolean isSecure() {
+			return this.responseCookie.isSecure();
+		}
+
+		@Nullable
+		@Override
+		public SameSite getSameSite() {
+			// Adding non-null return site breaks tests.
+			return null;
+		}
+
+		@Override
+		public boolean isHttpOnly() {
+			return this.responseCookie.isHttpOnly();
+		}
+
+		@Override
+		public boolean isPartitioned() {
+			return false;
+		}
+
+		@Override
+		public Map<String, String> getAttributes() {
+			return Collections.emptyMap();
+		}
+	}
+
+	private static class ContentWriterIteratingCallback extends IteratingCallback {
+		private final SeekableByteChannel source;
+
+		private final Content.Sink sink;
+
+		private final Callback callback;
+
+		private final RetainableByteBuffer buffer;
+
+		private final long length;
+
+		private long totalRead = 0;
```

Comment:
Good idea.  It might be in a slightly different form... probably something like:
```java
			Content.copy(Content.Source.from(channel, position, count), this.response, callback);
```
I'll try to get that in this months release.